src/Symfony/Component/Routing/RouteCollection.php

```diff
@@ -134,11 +134,11 @@ public function addCollection(self $collection)
     /**
      * Adds a prefix to the path of all child routes.
      *
-     * @param string $prefix       An optional prefix to add before each pattern of the route collection
-     * @param array  $defaults     An array of default values
-     * @param array  $requirements An array of requirements
+     * @param string|null $prefix       An optional prefix to add before each pattern of the route collection
+     * @param array       $defaults     An array of default values
+     * @param array       $requirements An array of requirements
      */
-    public function addPrefix($prefix, array $defaults = [], array $requirements = [])
+    public function addPrefix(?string $prefix, array $defaults = [], array $requirements = [])
```

Comment:
do we really need to make the prefix nullable? i would prefer if not. the prefix maps to the path which is not nullable. so this seems inconsistent. for `setHost` it's different because the host on the Route itself is also nullable.

Reply:
There are ten unit tests that pass `null` as `$prefix`. Apart from that, I agree – especially since the method is a no-op when called with `null`.

Reply:
Okay, actually we need this to be nullable because of the way `RouteCollectionBuilder` interacts with `RouteCollection`.

https://github.com/symfony/symfony/blob/1d57b800aabdbe1ae23f9e358bf2395ac24578dc/src/Symfony/Component/Routing/RouteCollectionBuilder.php#L312

This saves us a if block there. 🤷‍♂

Reply:
please add a string cast in the RouteCollectionBuilder `$subCollection->addPrefix((string) $this->prefix); ` and remove nullable from `addPrefix` which was not documented

Reply:
nvm, keep the nullable here and we could still trigger a deprecation in 4.4 if prefix is null in separate step.

Reply:
See #32246 for the deprecation.